src/ipa_pc/mod.rs

```diff
@@ -789,89 +938,112 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
+        let batch_check_time = start_timer!(|| "Multi poly multi point check batching");
 
+        let mut query_to_labels_map = BTreeMap::new();
         for (label, (point_label, point)) in query_set.iter() {
             let labels = query_to_labels_map
                 .entry(point_label)
                 .or_insert((point, BTreeSet::new()));
             labels.1.insert(label);
         }
 
-        assert_eq!(proof.len(), query_to_labels_map.len());
+        // v_i values
+        let mut v_values = batch_proof.batch_values.clone();
 
-        let mut randomizer = G::ScalarField::one();
+        // y_i values
+        let mut y_values = vec![];
 
-        let mut combined_check_poly = Polynomial::zero();
-        let mut combined_final_key = <G::Projective as ProjectiveCurve>::zero();
+        // x_i values
+        let mut points = vec![];
 
-        for ((_point_label, (point, labels)), p) in query_to_labels_map.into_iter().zip(proof) {
-            let lc_time =
-                start_timer!(|| format!("Randomly combining {} commitments", labels.len()));
-            let mut comms: Vec<&'_ LabeledCommitment<_>> = Vec::new();
-            let mut vals = Vec::new();
+        for (_point_label, (point, labels)) in query_to_labels_map.into_iter() {
             for label in labels.into_iter() {
-                let commitment = commitments.get(label).ok_or(Error::MissingPolynomial {
-                    label: label.to_string(),
-                })?;
-
-                let v_i = values
+                let y_i = evaluations
                     .get(&(label.clone(), *point))
                     .ok_or(Error::MissingEvaluation {
                         label: label.to_string(),
                     })?;
-
-                comms.push(commitment);
-                vals.push(*v_i);
+                y_values.push(*y_i);
+                points.push(point);
             }
+        }
 
-            let check_poly = Self::succinct_check(
-                vk,
-                comms.into_iter(),
-                *point,
-                vals.into_iter(),
-                p,
-                opening_challenges,
-            );
+        // Commitment of the h(X) polynomial
+        let batch_commitment = batch_proof.batch_commitment;
 
-            if check_poly.is_none() {
-                return Ok(false);
-            }
 
-            let check_poly =
-                Polynomial::from_coefficients_vec(check_poly.unwrap().compute_coeffs());
-            combined_check_poly += (randomizer, &check_poly);
-            combined_final_key += &p.final_comm_key.into_projective().mul(&randomizer);
+        // TODO: When we will move to Sponge-based construction, we will absorb the batch_commitment only and then squeeze a new challenge
+        // Fresh random challenge x
+        let point = Self::compute_random_oracle_challenge(
+            &to_bytes![
+                batch_commitment,
+                points                
+            ]
```

Comment:
As in the prover procedure: hash the batch_commitment and all the previously used challenges lambda_i if we keep with a ``opening_challenge()`` for prover and verifier.